tasks/kmt.py

```diff
@@ -375,3 +376,83 @@ def clean(ctx, stack=None, container=False, image=False):
         ctx.run("docker rm -f $(docker ps -aqf \"name=kmt-compiler\")")
     if image:
         ctx.run("docker image rm kmt:compile")
+
+
+@task(
+    help={
+        "stacks": "Comma separated list of stacks to generate ssh config for. 'all' to generate for all stacks.",
+        "ddvm_rsa": "Path to the ddvm_rsa file to use for connecting to the VMs. Defaults to the path in the ami-builder repo",
+    }
+)
+def ssh_config(_, stacks=None, ddvm_rsa="~/dd/ami-builder/scripts/kernel-version-testing/files/ddvm_rsa"):
+    """
+    Print the SSH config for the given stacks.
+
+    Recommended usage: inv kmt.ssh-config --stacks=all > ~/.ssh/config-kmt.
+    Then add the following to your ~/.ssh/config:
+            Include ~/.ssh/config-kmt
+
+    This makes it easy to use the SSH config for all stacks whenever you change anything,
+    without worrying about overriding existing configs.
+    """
+    stacks_dir = Path(get_kmt_os().stacks_dir)
+    stacks_to_print = None
+
+    if stacks is not None and stacks != 'all':
+        stacks_to_print = set(stacks.split(','))
+
+    for stack in stacks_dir.iterdir():
+        if not stack.is_dir():
+            continue
+
+        output = stack / "stack.output"
+        if not output.exists():
+            continue  # Invalid/removed stack, ignore it
+
+        stack_name = stack.name.replace('-ddvm', '')
+        if (
+            stacks_to_print is not None
+            and 'all' not in stacks_to_print
+            and stack_name not in stacks_to_print
+            and stack.name not in stacks_to_print
+        ):
+            continue
+
+        metal_instances = {}
+        vms = []
+
+        with output.open() as f:
+            for line in f:
+                name, ip = line.strip().split(' ')
+
+                if name.endswith("-instance-ip"):  # This is the metal instance
+                    arch = name.replace('-instance-ip', '')
+                    metal_instances[arch] = ip
+                else:  # A VM inside a corresponding metal instance
+                    name_parts = name.split('-')
+                    if len(name_parts) < 2:
+                        raise Exit(f"Invalid machine name {name} in stack {stack}")
+
+                    arch = name_parts[0]
+                    name = name_parts[1]
+                    vms.append((arch, name, ip))
```

Comment:
You can use the functions `get_domain_name_and_ip` and `get_instance_ip` to parse the `stack.outputs` file.

Reply:
I didn't see them before, however they seem more geared towards search a specific entry rather than parsing the entire file right?

Reply:
That is true. It might be worth it to extract a common function for the file and use that in the functions I mentioned above as well. I imagine the layout of this file changing in the future, so that would make updating the parsing logic easier.

Reply:
Done! I created a new dataclass for metal instances, in the future we could make that and LibvirtDomain use the same code but I think for now it works.